### PR TITLE
fix(CachedManager): allow overriding constructor for makeCache

### PR DESCRIPTION
**Pull request by ckohen**
**Please describe the changes this PR makes and why it should be merged:**

I think this has only been breaking for MessageManager and ThreadManager. It probably applies to others as well though long term.

fixes: #9759 

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Review by Jiralite — APPROVED** [review]

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -4717,17 +4717,23 @@ export interface Caches {
...
   GuildMemberManager: [manager: typeof GuildMemberManager, holds: typeof GuildMember];
   GuildBanManager: [manager: typeof GuildBanManager, holds: typeof GuildBan];
-  GuildForumThreadManager: [manager: typeof GuildForumThreadManager, holds: typeof ThreadChannel];
+  GuildForumThreadManager: [
+    manager: typeof GuildForumThreadManager,
+    holds: typeof ThreadChannel<true>,
+    overriden: true,
+  ];
   GuildInviteManager: [manager: typeof GuildInviteManager, holds: typeof Invite];
+  GuildMessageManager: [manager: typeof GuildMessageManager, holds: typeof Message<true>];
```
Comment (jaw0r3k) [inline packages/discord.js/typings/index.d.ts:4733]:
```suggestion 
  GuildMessageManager: [manager: typeof GuildMessageManager, holds: typeof Message<true>, overriden: true];
```

File: packages/discord.js/typings/index.d.ts
Comment (sdanialraza) [inline packages/discord.js/typings/index.d.ts:1]: I see `overriden` a lot here, isn't it overrid**d**en?

File: packages/discord.js/typings/index.d.ts [hunk 2]
```diff
@@ -4717,17 +4717,23 @@ export interface Caches {
...
   GuildScheduledEventManager: [manager: typeof GuildScheduledEventManager, holds: typeof GuildScheduledEvent];
   GuildStickerManager: [manager: typeof GuildStickerManager, holds: typeof Sticker];
-  GuildTextThreadManager: [manager: typeof GuildTextThreadManager, holds: typeof ThreadChannel];
+  GuildTextThreadManager: [manager: typeof GuildTextThreadManager, holds: typeof ThreadChannel<false>, overriden: true];
```
Comment (jaw0r3k) [inline packages/discord.js/typings/index.d.ts:4736]:
```suggestion 
  GuildTextThreadManager: [manager: typeof GuildTextThreadManager, holds: typeof ThreadChannel<false>];

**Review by sdanialraza — APPROVED** [review]

**Review by jaw0r3k — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by kevinlul** [comment]
Thanks for fixing and releasing. This was probably the cause memory leaks in our bot.

**Pull request merged**